### ISSUE TITLE
MASSEFFECT2.EXE_0xFCCCBE5F.dds - unhandled exception - Index out of range

### ISSUE THREAD

**Issue by Thommaal** [MassEffectModder/MassEffectModder#3]
![masseffectmodder_issue04](https://cloud.githubusercontent.com/assets/24566859/21219775/72210844-c2b6-11e6-8c7a-a09d7704f3c7.png)

"MASSEFFECT2.EXE_0xFCCCBE5F.dds" is a file in the ME2 ALOT [Update 2.5.11](http://www.nexusmods.com/masseffect2/mods/68/?tab=2&navtag=http%3A%2F%2Fwww.nexusmods.com%2Fmasseffect2%2Fajax%2Fmodfiles%2F%3Fid%3D68&pUp=1). 

MassEffectModder crashes when it tries to apply a mem that includes this file. Creating and loading a mem with the dds inside works without a hitch.

Detailed error report:
`See the end of this message for details on invoking 
just-in-time (JIT) debugging instead of this dialog box.

************** Exception Text **************
System.ArgumentOutOfRangeException: Index was out of range. Must be non-negative and less than the size of the collection.
Parameter name: index
   at System.ThrowHelper.ThrowArgumentOutOfRangeException(ExceptionArgument argument, ExceptionResource resource)
   at System.Collections.Generic.List`1.get_Item(Int32 index)
   at MassEffectModder.TexExplorer.replaceTexture(DDSImage image, List`1 list, String textureName) in c:\stuff\MassEffectModder\MassEffectModder\TexExplorerReplace.cs:line 124
   at MassEffectModder.TexExplorer.processTextureMod(String filenameMod, Int32 previewIndex, Boolean extract, Boolean replace, String outDir) in c:\stuff\MassEffectModder\MassEffectModder\TexExplorerProcess.cs:line 130
   at MassEffectModder.TexExplorer.replaceTextureMod(String filenameMod) in c:\stuff\MassEffectModder\MassEffectModder\TexExplorerProcess.cs:line 45
   at MassEffectModder.TexExplorer.applyModToolStripMenuItem_Click(Object sender, EventArgs e) in c:\stuff\MassEffectModder\MassEffectModder\TexExplorer.cs:line 499
   at System.Windows.Forms.ToolStripItem.RaiseEvent(Object key, EventArgs e)
   at System.Windows.Forms.ToolStripMenuItem.OnClick(EventArgs e)
   at System.Windows.Forms.ToolStripItem.HandleClick(EventArgs e)
   at System.Windows.Forms.ToolStripItem.HandleMouseUp(MouseEventArgs e)
   at System.Windows.Forms.ToolStrip.OnMouseUp(MouseEventArgs mea)
   at System.Windows.Forms.ToolStripDropDown.OnMouseUp(MouseEventArgs mea)
   at System.Windows.Forms.Control.WmMouseUp(Message& m, MouseButtons button, Int32 clicks)
   at System.Windows.Forms.Control.WndProc(Message& m)
   at System.Windows.Forms.ToolStrip.WndProc(Message& m)
   at System.Windows.Forms.ToolStripDropDown.WndProc(Message& m)
   at System.Windows.Forms.NativeWindow.Callback(IntPtr hWnd, Int32 msg, IntPtr wparam, IntPtr lparam)

************** Loaded Assemblies **************
mscorlib
    Assembly Version: 4.0.0.0
    Win32 Version: 4.6.1586.0 built by: NETFXREL2
    CodeBase: file:///C:/Windows/Microsoft.NET/Framework64/v4.0.30319/mscorlib.dll
----------------------------------------
MassEffectModder
    Assembly Version: 1.0.0.0
    Win32 Version: 1.0.0.0
    CodeBase: file:///H:/Aleph/ME3/Werkzeuge/MassEffectModder/MassEffectModder.exe
----------------------------------------
System.Windows.Forms
    Assembly Version: 4.0.0.0
    Win32 Version: 4.6.1586.0 built by: NETFXREL2
    CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.Windows.Forms/v4.0_4.0.0.0__b77a5c561934e089/System.Windows.Forms.dll
----------------------------------------
System
    Assembly Version: 4.0.0.0
    Win32 Version: 4.6.1586.0 built by: NETFXREL2
    CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System/v4.0_4.0.0.0__b77a5c561934e089/System.dll
----------------------------------------
System.Drawing
    Assembly Version: 4.0.0.0
    Win32 Version: 4.6.1586.0 built by: NETFXREL2
    CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.Drawing/v4.0_4.0.0.0__b03f5f7f11d50a3a/System.Drawing.dll
----------------------------------------
System.Configuration
    Assembly Version: 4.0.0.0
    Win32 Version: 4.6.1586.0 built by: NETFXREL2
    CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.Configuration/v4.0_4.0.0.0__b03f5f7f11d50a3a/System.Configuration.dll
----------------------------------------
System.Core
    Assembly Version: 4.0.0.0
    Win32 Version: 4.6.1586.0 built by: NETFXREL2
    CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.Core/v4.0_4.0.0.0__b77a5c561934e089/System.Core.dll
----------------------------------------
System.Xml
    Assembly Version: 4.0.0.0
    Win32 Version: 4.6.1586.0 built by: NETFXREL2
    CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.Xml/v4.0_4.0.0.0__b77a5c561934e089/System.Xml.dll
----------------------------------------
StreamHelpers
    Assembly Version: 1.0.0.0
    Win32 Version: 1.0.0.0
    CodeBase: file:///H:/Aleph/ME3/Werkzeuge/MassEffectModder/StreamHelpers.DLL
----------------------------------------
ZlibHelper
    Assembly Version: 1.0.0.0
    Win32 Version: 1.0.0.0
    CodeBase: file:///H:/Aleph/ME3/Werkzeuge/MassEffectModder/ZlibHelper.DLL
----------------------------------------
LZO2Helper
    Assembly Version: 1.0.0.0
    Win32 Version: 1.0.0.0
    CodeBase: file:///H:/Aleph/ME3/Werkzeuge/MassEffectModder/LZO2Helper.DLL
----------------------------------------

************** JIT Debugging **************
To enable just-in-time (JIT) debugging, the .config file for this
application or computer (machine.config) must have the
jitDebugging value set in the system.windows.forms section.
The application must also be compiled with debugging
enabled.

For example:

<configuration>
    <system.windows.forms jitDebugging="true" />
</configuration>

When JIT debugging is enabled, any unhandled exception
will be sent to the JIT debugger registered on the computer
rather than be handled by this dialog box.
`

**Comment by Thommaal** [comment]
I have added the mem within a zip; it only contains the dds in question.
[MASSEFFECT2.EXE_0xFCCCBE5F.zip](https://github.com/MassEffectModder/MassEffectModder/files/654349/MASSEFFECT2.EXE_0xFCCCBE5F.zip)

**Comment by aquadran** [comment]
looks like issue with DDS fille. process of creating MOD didn't catch issue. I'll check it when back home

**Comment by aquadran** [comment]
btw. you don't need zip Mod file, it's already compressed :)

**Comment by Thommaal** [comment]
Yeah, I tried it at first; but the mem-extension is not supported: it needs to be a PNG, GIF, JPG, DOCX, PPTX, XLSX, TXT, PDF, or ZIP. Well, that's what the system message is telling me.

Thanks for looking into it; at least, it could be a genuine issue this time, and not just a botched installation. :)

**Comment by aquadran** [comment]
it's internally zippped

**Comment by aquadran** [comment]
I tried this mod, works fine for me. Can you try apply only on vanilla, I wonder if it's somehow correllated to other replaced textures

**Comment by alvarome** [comment]
Hi:

@Thommaal I have downloaded your .mem file and I have just installed it flawlessly. 

So I can confirm it is also working fine at my end.

HTH!
Alvaro

**Comment by Thommaal** [comment]
And I can confirm that the mem with the problematic dds inside installs perfectly on a Vanilla ME2. Which brings us back to aquadran's hypothesis, that some other file or files that were changed by ALOT are involved.

**Comment by Thommaal** [comment]
Maybe it's better to tell you what I did exactly:
Since it was the first time that I used MassEffectModder, I made mems out of every single subfolder within the ALOT structure and loaded and applied them one by one except for the tiny character mems, that I did in just two steps.

After ALOt was completely done, I made a mem out of the update file , and tried to apply this one last. It worked fine till it reached the mentioned dds that caused a crash.

I excluded the file from the update pack, made a mem and applied that one without a hitch. Then I made a mem out of the problematic file, but it crashed again.

**Comment by aquadran** [comment]
ok, I'll try repeat yours steps

**Comment by alvarome** [comment]
@Thommaal I´ve been using MEM for both ME2 and ME3 modding runs in the very same way: 35 .mem files for ME2 // 46 .mem files for ME3 but I´m not having this issue here.

Hard to tell what´s going on here but, as you said, it could be related to other textures previously installed.

BTW... what textures are those "tiny characters mems if I can ask? Just out of curilsity.

**Comment by Thommaal** [comment]
Just to repeat my addition (so that it doesn't get lost after you have answered so quickly): IF I add the entire Update as a mem to ME2, it crashes.

@alvarome, I was refering to the characters in the Squad folder: Ashley, Garrus, etc.

**Comment by alvarome** [comment]
Ok, thanks! 

BTW, your replies are rather clear, no problem I think! :) 

Looks like your issue could be triggered also when installing latest update textures as an unique .mem file then... weird! No idea here.

**Comment by Thommaal** [comment]
Yeah, weird indeed. So, I am going to Vanilla ME 2 again, add all the ALOT-mems, and then I will add just the problem-dds and finally the Update-mem that excludes this one file. I am pretty sure, it's going to work in this sequence.

**Comment by alvarome** [comment]
Wait! Really? Just for one texture? 

BTW, may you please check if that offending texture is already at the ALOT "base" mod files? 

Just ot know if the texture is installed twice... a newer version onto an older one. I only have latest in my setup...

**Comment by Thommaal** [comment]
Yes, it's in Environment > Normandy. 
[MASSEFFECT2.EXE_0xFCCCBE5F.zip](https://github.com/MassEffectModder/MassEffectModder/files/655867/MASSEFFECT2.EXE_0xFCCCBE5F.zip)

**Comment by alvarome** [comment]
Ok... that at Normandy is 4096x4096 while the one in the update is 1024x1024 so we are effectively "downscaling" an installed texture here a you first install the larger and then the smaller. 

Just sharing info here.... no idea if that could be related in any way to this issue at all... -?-

**Comment by aquadran** [comment]
I can reproduse issue

**Comment by alvarome** [comment]
Excellent! Bug confirmed! :)

**Comment by Thommaal** [comment]
At that stage, it is indeed excellent news.  Okay, then I'm going to re-integrate my ALOT-memed ME2 into my Steam folder, and see what happens when I add a couple of Ottemises babies.

**Comment by alvarome** [comment]
Good work Thommaal! :)

Enjoy and ensure to delete those star textures! ;)

**Comment by Thommaal** [comment]
You're welcome. I will check in again later to see if a solution came up. If there is anything that needs testing, aquadran, I have a vanilla ME2 ready, and I will also backup the ALOT-mem-ME2 prior to any other texture files I might add.

**Comment by alvarome** [comment]
Knowing how much, good and fast Aquadran use to work, I would expect a new MEM release anytime... :)

**Comment by aquadran** [comment]
you can replace that texture from update to main. issue seems related to downscalling

**Comment by aquadran** [comment]
ok, it seems I fixed issue, however did not tested full run from ALOT.
So please grab 1.22 release.

**Comment by Thommaal** [comment]
It works with v1.22.

I made a mem from the entire update file (including the troublesome dds), and had no poblem to apply it on an ALOT-memed ME2. Then I made a mem from the overwritten dds that had been in ALOT > Environment > Normandy, and applied it over ME2 without a hitch after the update.mem had done its work. Redoing the update-file-mem worked as well.

In case the texture modders need to avoid some specific procedure to make sure that downgraded files do not cause crashes, I'd suggest to publish instructions (e.g. within the readme for MassEffectModder).

And since it looks like you solved the problem, I'd vote for closing the issue. :) Thank you.
